File: src/EnvironmentMonitor.Domain/Enums/MeasurementSourceTypes.cs
using System.ComponentModel;

namespace EnvironmentMonitor.Domain.Enums
{
    public enum MeasurementSourceTypes
    {
        [Description("IoT Hub")]
        IotHub = 0,
        [Description("Rest interface")]
        Rest = 1,
        [Description("Ilmatieteenlaitos Open Data")]
        Ilmatieteenlaitos = 2
    }
}

File: src/EnvironmentMonitor.ReactUi/environment-monitor/src/enums/measurementSourceTypes.ts
export enum MeasurementSourceTypes {
  IotHub = 0,
  Rest = 1,
  Ilmatieteenlaitos = 2,
}

export const getMeasurementSourceDisplayName = (
  sourceId: MeasurementSourceTypes,
): string => {
  switch (sourceId) {
    case MeasurementSourceTypes.IotHub:
      return "IoT Hub";
    case MeasurementSourceTypes.Rest:
      return "Rest interface";
    case MeasurementSourceTypes.Ilmatieteenlaitos:
      return "Ilmatieteen laitos (Open data)";
    default:
      return "Unknown";
  }
};
